fiat-ldap/src/test/groovy/com/netflix/spinnaker/fiat/roles/ldap/LdapUserRolesProviderTest.groovy

```diff
@@ -284,6 +285,35 @@ class LdapUserRolesProviderTest extends Specification {
     roles == [user1: [role1, role2, role3]]
   }
 
+  @Unroll
+  void "loadRolesForUser should merge roles when multiple DNs exist for the user id"() {
+    given:
+    def user = externalUser("user1")
+    def role1 = new Role("group1").setSource(Role.Source.LDAP)
+    def role2 = new Role("group2").setSource(Role.Source.LDAP)
+
+    def configProps = baseConfigProps()
+
+    def provider = Spy(LdapUserRolesProvider) {
+      getPartialUserDn(_ as String) >> null
+      getUserDNs(_ as Collection<String>) >> [ 'uid=dn1,ou=users,dc=springframework,dc=org' : 'user1',
+                                               'uid=dn2,ou=users,dc=springframework,dc=org' : 'user1']
+    }
+    provider.ldapTemplate = Mock(SpringSecurityLdapTemplate) {
+      _ * searchForSingleEntry(*_) >> { throw new IncorrectResultSizeDataAccessException(1) }
```

Comment:
```suggestion
      1 * searchForSingleEntry(*_) >> { throw new IncorrectResultSizeDataAccessException(1) }
```
or whatever the actual number is, to clarify the interactions with the provider.

Reply:
Done